test.py

```diff
@@ -1326,11 +1326,16 @@ def prepare_dir(dirname, pattern):
 
 async def find_tests(options: argparse.Namespace) -> None:
 
-    for f in glob.glob(os.path.join("test", "*")):
-        if os.path.isdir(f) and os.path.isfile(os.path.join(f, "suite.yaml")):
-            for mode in options.modes:
-                suite = TestSuite.opt_create(f, options, mode)
-                await suite.add_test_list()
+    suite_names = [tc.suite_name for tc in options.names]
```

Comment:
Overall I like the idea of saving on the load/preprocess time if wildcard is specified very match. But I think it kills the idea of a general purpose pattern that can be specified for 'names'. E.g. if I type ./test.py raft:: is it selecting a single suite or a single test file with a matching name in all suites? It would be lovely to have a powerful filter capabilities and fast search for the matching tests, something I can't think of a way of combining in a single pattern match.

Reply:
I wrote in another review comment:

>>> What would you think of `./test.py raft` for matching any and `./test.py raft::` for matching the suite raft? We need to make it unambiguous.